kubemarine/kubernetes_accounts.py

```diff
@@ -54,7 +55,49 @@ def enrich_inventory(cluster: KubernetesCluster) -> None:
             rbac["accounts"][i]['configs'][2]['metadata']['namespace'] = account['namespace']
 
 
+def handle_authenticated_sa_issuer_discovery(cluster: KubernetesCluster) -> None:
+    """
+    This function handles SA issuer discovery endpoint authentication
+
+    To disable authentication, we apply CRB which allows unauthenticated access.
+    To enable authentication, we delete above CRB (if it is present).
+
+    For more information on the cluster role: 
+    https://kubernetes.io/docs/tasks/configure-pod-container/configure-service-account/#service-account-issuer-discovery
+    """
+    mode = "unauthenticated"
+    kubectl_cmd = "apply"
+    if cluster.inventory.get("rbac", {}).get("authenticated-issuer-discovery", False):
+        mode = "authenticated"
+        kubectl_cmd = "delete --ignore-not-found"
+    issuer_discovery_crb = dedent("""\
+        kind: ClusterRoleBinding
```

Comment:
Maybe we could put it into `defaults.yaml` like that one: https://github.com/Netcracker/KubeMarine/blob/99f13807ed0f79bc9448c371eeb2951e15466b9b/kubemarine/resources/configurations/defaults.yaml#L868

Reply:
We could, if we want this CRB to be customizable by user, but I think we do not need that. Also, the less parameters we expose to the user, the less to support. I suggest to keep this CRB hidden in code, if possible

Reply:
Today we need to allow unauthenticated access to the one endpoint, tomorrow it will need another one. It's much better to have generalized approach.  And let's user customize that part of configuration.

Reply:
For users, we have generalized approach - plugins. Unfortunately, for our usecase plugins do not work in its current state - they do not support "uninstall". They only support "install", or nothing. We need "uninstall" to allow authenticated access. This requires enhancement to plugins, which is out of scope for this feature I believe